app/src/forms/common/scheduleService.js

```diff
@@ -0,0 +1,597 @@
+const moment = require('moment-timezone');
+const { ScheduleType, EmailTypes } = require('./constants');
+
+/**
+ * Core date calculation functions
+ * These don't know about schedule objects, just work with raw dates
+ */
+
+/**
+ * Check if a date is in the future
+ * @param {String|Date} date The date to check
+ * @param {String} timezone The timezone to evaluate in
+ * @param {String|Date} referenceDate Optional reference date (defaults to now)
+ * @param {Boolean} ignoreTime Whether to ignore time component (defaults to true)
+ * @returns {Boolean} True if the date is in the future
+ */
+function isDateInFuture(date, timezone = 'America/Vancouver', referenceDate = null, ignoreTime = true) {
```

Comment:
Make 'America/Vancouver' and the date formats ('YYYY-MM-DD') into constants. Maybe in the future we can pull them from configuration... but for now, have them declared once so we can change them easily.